pkg/datastore/datastore.go

```diff
@@ -217,19 +230,19 @@ func (ds *memoryDatastore) GetPackageIDsByOperatorSource(opsrcUID types.UID) str
 	return strings.Join(packages, ",")
 }
 
-func (ds *memoryDatastore) CheckPackages(packageIDs []string) error {
+func (ds *memoryDatastore) CheckPackages(source string, packageIDs []string) error {
 	missingPackages := []string{}
 	for _, packageID := range packageIDs {
-		if _, err := ds.Read(packageID); err != nil {
+		if _, err := ds.Read(source, packageID); err != nil {
 			missingPackages = append(missingPackages, packageID)
 			continue
 		}
 	}
 
 	if len(missingPackages) > 0 {
 		return fmt.Errorf(
-			"Still resolving package(s) - %s. Please make sure these are valid packages.",
-			strings.Join(missingPackages, ","),
+			"Still resolving package(s) - %s. Please make sure these are valid packages within the %s source.",
```

Comment:
This error message seems like it could be confusing in a case where the user didn't specify the source themselves. Can we we be more explicit that the issue is with the `spec.Source` field? So: `Please make sure these are valid packages within the %s spec.Source`?

Reply:
If the source was resolved by marketplace, this code wouldn't be hit given that the resolved source would contain all the packages, but I will update the message to improve clarity.